Tasks/DeployAzureResourceGroup/Environment.ts

```diff
@@ -204,5 +203,124 @@ export class RegisterEnvironment {
+                        this._addExtensionVM(vmName, computeClient, dnsName, _extensionName, location);
+                    }
+                    else{
+                        //Validate the Custom Script Execution status: if ok add the rule else add the extension
+                        this.Validate_CustomScriptExecutionStatus(vmName, computeClient, dnsName, _extensionName, location);
+                    }
+                }
+            });
+        }
+        catch(exception){
+            throw new Error("ARG_DeploymentPrereqFailed" + exception.message);
+        }
+    }
+
+    private Validate_CustomScriptExecutionStatus(vmName: string, computeClient, dnsName: string, extensionName: string, location: string){
+        console.log("Validating the winrm configuration custom script extension status");
+
+        computeClient.virtualMachines.get(this.resourceGroupName, vmName, {expand: 'instanceView'}, (error, result, request, response)=>{
+            if(error){
+                console.log("Error in getting the instance view of the virtual machine %s", util.inspect(error, {depth: null}));
+                throw new Error("FailedToFetchInstanceViewVM");
+            }
+            console.log(util.inspect(result, {depth: null}));
+            var invalidExecutionStatus: boolean = false;
+            var extensions = result["instanceView"]["extensions"];
+            for(var i =0; i < extensions.length; i++){
+                var extension = extensions[i];
+                if(extension["name"] === extensionName){
+                    for(var j =0; j < extension["substatuses"]; j++){
+                        var substatus = extension["substatuses"][j];
+                        if(substatus["code"].include("ComponentStatus/StdErr") && !!substatus["message"] && substatus["message"] != ""){
+                            invalidExecutionStatus = true;
+                            break;
+                        }
+                    }
+                }
+            }
+            if(invalidExecutionStatus){
+                this._removeExtensionFromVM(extensionName, vmName, computeClient);
+                this._addExtensionVM(vmName, computeClient, dnsName, extensionName, location);
+            }
+            else{
+                this.Add_WinRMHttpsNetworkSecurityRuleConfig(vmName);
+            }
+        });
+    }
+
+    private _addExtensionVM(vmName: string, computeClient, dnsName: string, extensionName: string, location: string){
+        var _configWinRMScriptFile: string ="https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/201-vm-winrm-windows/ConfigureWinRM.ps1";
+        var _makeCertFile: string ="https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/201-vm-winrm-windows/makecert.exe";
+        var _winrmConfFile: string ="https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/201-vm-winrm-windows/winrmconf.cmd";
```

Comment:
Are you sure about using raw.github links? 
What happens when there are any breaking changes?
Perhaps, you could stick to one commit ID